File: learn-algo/leetcode/709/go/709.go
package main

import "fmt"

func toLowerCase(s string) string {
	str := []byte(s)
	for i := 0; i < len(s); i++ {
		if str[i] <= 'Z' && str[i] >= 'A' {
			str[i] += 'a' - 'A'
		}
	}

	return string(str)
}

func main() {
	s := "AaBbCc"
	res := toLowerCase(s)
	fmt.Println(res)
}

File: learn/learn-algo/leetcode/709/c++/709.cpp
#include <iostream>
#include <algorithm>
#include <string>
#include <vector>

using namespace std;

class Solution {
public:
    string toLowerCase(string str) {
        transform(str.begin(), str.end(), str.begin(), ::tolower);
        return str;
    }
};

int main() {
    Solution s;
    vector<int> in = {1, 2, 1, 2, 6, 7, 5, 1};
    string str = "Hello";
    auto res = s.toLowerCase(str);
    for_each(res.begin(), res.end(), [](auto t) { cout << t ; });
    cout << endl;
    return 0;
}